lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3167,14 +3168,38 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
     attr->setVJPFunction(vjp);
   }
 
-  auto insertion =
-      ctx.DifferentiableAttrs.try_emplace({D, checkedWrtParamIndices}, attr);
+  auto insertion = ctx.DifferentiableAttrs.try_emplace(
+      {D, attr->getParameterIndices()}, attr);
   // `@differentiable` attributes are uniqued by their parameter indices.
   // Reject duplicate attributes for the same decl and parameter indices pair.
   if (!insertion.second && insertion.first->getSecond() != attr) {
-    diagnoseAndRemoveAttr(attr, diag::differentiable_attr_duplicate);
+    TC.diagnose(attr->getLocation(), diag::differentiable_attr_duplicate);
+    TC.diagnose(insertion.first->getSecond()->getLocation(),
+                diag::differentiable_attr_duplicate_note);
     return;
   }
+  // Transfer `@differentiable` attribute from storage declaration to
+  // getter accessor.
+  if (auto *asd = dyn_cast<AbstractStorageDecl>(D)) {
```

Comment:
Is it possible to use the original attribute instead? Just set `implicit` to `true`.

Reply:
Is that desirable? The original attribute isn't implicit though - I think setting `implicit = true` may affect diagnostics/dumping.

Reply:
In the new attribute you created in line 3185, it's also `implicit`. Did that make a difference to diagnostics or no?

Reply:
I'd suggest writing a computed property with a non-differentiable body, mark the property decl as `@differentiable`, and see what error the AD pass emits.

Reply:
Aha - since the original attribute is removed the storage declaration, there's no reason to not mutate/reuse it.

Done in https://github.com/apple/swift/pull/25293/commits/825066220d45c05a58afd3556b78d77f27f234ce.